### ISSUE TITLE
Using include creates false positive UnusedVariable

### ISSUE THREAD

**Issue by mathe42** [vimeo/psalm#10324]
Example:

```
<?
// file __1.php
$a = 1;
include("./__2.php");
```

```
<?
// file __2.php
if($a) {
  echo "a is true";
}
```

This gives me for __file2.php a UndefinedGlobalVariable that is ok as psalm has no context for that. 
But it gives me in __1.php a UnusedVariable "$a is never referenced or the value is not used" Error.

Note:
In our own custom framework this pattern is used to an extend that a supress comment for each case is not a good option...

**Comment by mathe42** [comment]
I cant reproduce this on https://psalm.dev/ as it includes 2 files.

**Comment by steal3rd** [comment]
I can confirm this behavior. I experience the same situation with a function that sets a number of variables, then includes another file as a template to render. Those variables are marked as:

> [UnusedVariable] $variableName is never referenced or the value is not used psalm (77)

The included file references the variables, but those references are marked as:

> [UndefinedGlobalVariable] Cannot find referenced variable $variableName in global scope psalm (127)

```php
<?php

// Class for rendering notification messages.

class Notifications
{
	public function error() : void
	{
		$message = "Error message.";
		
		include __DIR__ . "/templates/error.php";
	}
	
	public function info() : void
	{
		$message = "Informational message.";
		
		include __DIR__ . "/templates/info.php";
	}
	
	public function warning() : void
	{
		$message = "Warning message.";
		
		include __DIR__ . "/templates/warning.php";
	}
}
```
The template might be something like this.

```php
<div class="notification"><?= $message ?></div>
```

The reason why this happens is appreciated. I hope this example helps illustrate what I believe the original poster is experiencing.
